compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -325,7 +325,10 @@ impl<'tcx> Printer<'tcx> for V0SymbolMangler<'tcx> {
             || &args[..generics.count()]
                 == self
                     .tcx
-                    .erase_regions(ty::GenericArgs::identity_for_item(self.tcx, impl_def_id))
+                    .erase_and_anonymize_regions(ty::GenericArgs::identity_for_item(
+                        self.tcx,
+                        impl_def_id,
+                    ))
```

Comment:
:thinking: can you change the assert in line 341 to be `!args.has_non_region_param() && !args.has_free_regions()`?